payment-handler/can-make-payment-event.https.html

```diff
@@ -0,0 +1,98 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Tests for CanMakePaymentEvent</title>
+<link rel="help" href="https://w3c.github.io/payment-handler/#the-canmakepaymentevent">
+<link rel="manifest" href="manifest.json">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+async function registerAppAndBuildPaymentRequest(methodName) {
+  await navigator.serviceWorker.register('app-can-make-payment.js');
+  registration = await navigator.serviceWorker.ready;
+  await registration.paymentManager.instruments.set('instrument-key', {
+    name: 'Test Payment Method',
+    enabledMethods: [methodName],
+  });
+  const request = new PaymentRequest([{
+    supportedMethods: methodName,
+  }], {
+    total: {
+      label: 'Total',
+      amount: {
+        currency: 'USD',
+        value: '0',
+      },
+    },
+  });
+  return request;
+}
+
+promise_test(async t => {
+  const methodName = window.location.origin + '/canMakePayment-false';
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult = false;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(err.name, 'NotAllowedError', 'If it throws, then it must be NotAllowedError');
+  }
+  assert_false(paymentRequestCanMakePaymentResult, 'canMakePayment() should return false.');
+  await promise_rejects(t, 'NotSupportedError', request.show());
+}, 'If CanMakePaymentEvent.respondWith(false) is called, then the payment method is not supported.');
+
+promise_test(async t => {
+  const methodName = window.location.origin + '/canMakePayment-promise-false';
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult = false;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(err.name, 'NotAllowedError', 'If it throws, then it must be NotAllowedError');
+  }
+  assert_false(paymentRequestCanMakePaymentResult, 'canMakePayment() should return false.');
```

Comment:
if `catch` happens, would this be a false positive?

Reply:
If `catch` happens, then comparing `paymentRequestCanMakePaymentResult` would not make sense. Let's start with an undefined value and then set it to the expected value in the `catch`. This will prevent the test failing due to `NotAllowedError`.